polymer/src/sample-category-data.html

```diff
@@ -0,0 +1,168 @@
+<link rel="import" href="../node_modules/@npm-polymer/iron-ajax/iron-ajax.html">
+
+<dom-module id="sample-category-data">
+  <template>
+    <iron-ajax auto url="https://ampbyexample.com/sitemap.json" handle-as="json"
+      on-response="handleResponse"></iron-ajax>
+  </template>
+
+  <script>
+  (function() {
+
+    class SampleCategoryData extends Polymer.Element {
+
+      static get is() { return 'sample-category-data'; }
+
+      static get properties() { return {
+
+        categoryName: String,
+
+        itemName: String,
+
+        categories: {
+          type: Array,
+          notify: true
+        },
+
+        category: {
+          type: Object,
+          notify: true,
+          computed: '_computeCategory(categories, categoryName)',
+        },
+
+        item: {
+          type: Object,
+          computed: '_computeItem(category.items, itemName)',
+          notify: true
+        },
+
+        failure: {
+          type: Boolean,
+          notify: true,
+          readOnly: true
+        },
+
+        resource: {
+          type: Object
+        }
+
+      }}
+
+      handleResponse(data) {
+        this.resource = data.detail.response[0]["categories"]
+        var catNames =  this.resource.map(function(cat) {
+          var category = {
+            name:  cat["name"],
+            id: cat["id"]
+          };
+          return category;
+        });
+        this.categories = catNames;
+      }
+
+      _getCategory(categoryName, categories) {
+        if (categories === undefined || categoryName === "index.html") {
+          return null;
+        }
+        return categories.find(function(c) {
+          return c.id === categoryName
+        })
+      }
+
+      _computeCategory(categories, categoryName) {
+
+        // Fetch the items of the category. Note that the fetch is asynchronous,
+        // which means `category.items` may not be set initially (but that path
+        // will be notified when the fetch completes).
+        let category = this._getCategory(categoryName, this.categories);
+        this._fetchItems(category, 1);
+        return category;
+      }
+
+      _computeItem(items, itemName) {
+        if (!items || !itemName) {
+          return;
+        }
+        return items.find (function(item) {
+          return item.url.endsWith(itemName.concat('/'))
+        })
+      }
+
+      _parseCategoriesFromJson(e) {
+        return JSON.parse(e.target.responseText)[0]["categories"]
+      }
+
+      _fetchItems(category) {
+        this._setFailure(false);
+        // Only fetch the items of a category if it has not been previously set.
+        if (!category || category.items) {
+          return;
+        }
+
+        var categories = this.resource
+        categories = categories.filter(function(cat) {
+                return cat.name == category.name
+        });
+
+        var items =  categories[0]["examples"].map(function(itemJson) {
+          var item = {
```

Comment:
you could directly return the object here

Reply:
done